src/Folder.m

```diff
@@ -102,7 +102,10 @@ -(instancetype)initWithId:(NSInteger)newId parentId:(NSInteger)newIdParent name:
  */
 -(NSInteger)unreadCount
 {
-	return unreadCount;
+	@synchronized(self)
```

Comment:
if we define these as `atomic` properties, I don't think we'll need to manually `@synchronized` them.

Reply:
I think you are correct. This only needs to be done if the accessors are overridden.